### DISCUSSION TITLE
Embedded Types

### DISCUSSION THREAD

**Discussion by rehatkathuria** [prisma/prisma#21233] [Q&A] [answered]
### Question

I have the following:
https://gist.github.com/rehatkathuria/94b06f3dfc2dedd063a025e507e4921b

If I try to do this:

```js
const user = await prisma.user.findFirstOrThrow({
	where: { id: auth.user_id },
	include: {
		checkins: context.checkins,
		interestPreference: context.interest_preferences,
		conversations: context.conversations === true ? {
			include: { participants: context.conversation_participants },
		} : false,
		profile_assets: context.profile_assets,
	},
})
```

then the call sites don't pick up the `conversation.participants` regardless of what I pass into the parameters for `conversations: boolean`. However, with the code in the gist, typescript appropriately picks up the embedded type. 

Is this a typescript limitation?

**Comment by ludralph** [comment, marked as answer]
Hi @rehatkathuria 👋 

Thank you for raising this question.

In the code snippet you provided, where you conditionally include the conversations field as false, TypeScript doesn't infer the types as expected.  In your code: 

```js
conversations: context.conversations === true ? {
    include: { participants: context.conversation_participants },
} : false,
```
You have introduced a conditional that makes the type of `conversations` non-deterministic at the type-checking phase. So, TypeScript might not be able to infer the exact type of `conversations` because TypeScript's static type is not capable of understanding the runtime values and conditions for `conversation`. I will say this is expected behaviour and not a limitation.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by rehatkathuria** [comment]
I figured that would be the case, thank you for the clarification @ludralph
